lib/source_file.cpp

```diff
@@ -242,13 +265,29 @@ cstring SourceInfo::toSourceFragment() const {
     return InputSources::instance->getSourceFragment(*this);
 }
 
+cstring SourceInfo::toSourceFragment2() const {
+    return InputSources::instance->getSourceFragment2(*this);
+}
+
 cstring SourceInfo::toPositionString() const {
     if (!this->isValid())
         return "";
     SourceFileLine position = InputSources::instance->getSourceLine(this->start.getLineNumber());
     return position.toString();
 }
 
+cstring SourceInfo::toSourcePositionData(unsigned *outLineNumber,
+                                         unsigned *outColumnNumber) const {
+    SourceFileLine position = InputSources::instance->getSourceLine(this->start.getLineNumber());
```

Comment:
Generally the way to deal with this (upstream changes to master you want in your working branch that you've already made a pull request for) is to use

    git pull --rebase upstream master

(or possibly `origin master` if your local repo is clone of p4c.org/p4c with your fork added, rather than a clone of a fork, with p4c.org/p4c added as upstream)

Followed by

    git push -f

to update the pull request in place